### PR TITLE
feat: add block_sample_counts public property to RawBinFile and Reader

### PR DESCRIPTION
**Pull request by pmolfese**
RawBinFile and Reader had no public API for per-block sample counts. The only way to access them was via the private _blobs and signal_blocks internals.

Add `block_sample_counts` as a cached_property on both RawBinFile and Reader, following the same pattern as the existing sampling_rates, durations, and num_channels properties.

This is the only reliable way to detect the EGI PSG off-by-one bug, where the last PNS block contains one fewer sample than the corresponding EEG block. See `Reader.block_sample_counts` docstring for the detection pattern.

### CODE REVIEW

File: mffpy/raw_bin_files.py [hunk 1]
```diff
@@ -121,6 +121,16 @@ def block_start_idx(self) -> np.ndarray:
         return np.cumsum(
             [0]+self.signal_blocks['num_samples'])
 
+    @cached_property
+    def block_sample_counts(self) -> list:
```
Comment (damian5710) [inline mffpy/raw_bin_files.py:125]:
```suggestion
    def block_sample_counts(self) -> List[int]:
```

File: mffpy/reader.py [hunk 2]
```diff
@@ -208,6 +208,29 @@ def num_channels(self) -> Dict[str, int]:
             for fn, bin_file in self._blobs.items()
         }
 
+    @cached_property
+    def block_sample_counts(self) -> Dict[str, list]:
```
Comment (damian5710) [inline mffpy/reader.py:212]:
```suggestion
    def block_sample_counts(self) -> Dict[str, List[int]]:
```

**Review by damian5710 — APPROVED** [review]
Hi @pmolfese . Thanks for contributing to this project. The changes look good to me, so we should be able to merge this pull request once tests pass.

**Comment by pmolfese** [comment]
Good catches on the type-hint cleanups! Will try to add those more in the future. Let me know what you think of #133 and #135. Both are edge-cases but do a surprising number on breaking the `Reader()` import or when trying to parse PNS channels.
